docs/fsharp/language-reference/casting-and-conversions.md

```diff
@@ -1,41 +1,112 @@
-This can be useful when you have to copy an existing record, and possibly change some of the field values.
+Note that the enumeration casts work only if the underlying type of the enumeration is compatible with the type being converted. In the following code, the conversion fails to compile because of the mismatch between `int32` and `uint32`.
 
-Take for instance a newly created record.
+```fsharp
+// Error: types are incompatible
+let col4 : Color = enum 2u
+```
 
-[!code-fsharp[Main](../../../samples/snippets/fsharp/lang-ref-1/snippet1905.fs)]
+For more information, see [Enumerations](enumerations.md).
 
-If you were to update only on field of that record you could use the *copy and update record expression* like the following:
+## Casting Object Types
+Conversion between types in an object hierarchy is fundamental to object-oriented programming. There are two basic types of conversions: casting up (upcasting) and casting down (downcasting). Casting up a hierarchy means casting from a derived object reference to a base object reference. Such a cast is guaranteed to work as long as the base class is in the inheritance hierarchy of the derived class. Casting down a hierarchy, from a base object reference to a derived object reference, succeeds only if the object actually is an instance of the correct destination (derived) type or a type derived from the destination type.
 
-[!code-fsharp[Main](../../../samples/snippets/fsharp/lang-ref-1/snippet1906.fs)]
+F# provides operators for these types of conversions. The `:>` operator casts up the hierarchy, and the `:?>` operator casts down the hierarchy.
 
-## See Also
-[Records](records.md)
+### Upcasting
+In many object-oriented languages, upcasting is implicit; in F#, the rules are slightly different. Upcasting is applied automatically when you pass arguments to methods on an object type. However, for let-bound functions in a module, upcasting is not automatic, unless the parameter type is declared as a flexible type. For more information, see [Flexible Types](flexible-Types.md).
+
+The `:>` operator performs a static cast, which means that the success of the cast is determined at compile time. If a cast that uses `:>` compiles successfully, it is a valid cast and has no chance of failure at run time.
+
+You can also use the `upcast` operator to perform such a conversion. The following expression specifies a conversion up the hierarchy.
+
+**upcast** *expression*
+
+When you use the upcast operator, the compiler attempts to infer the type you are converting to from the context. If the compiler is unable to determine the target type, the compiler reports an error.
```

Comment:
I'd make sure upcast is always formatted as inline code so it's not translated later when get localization working. There's at least two not formatted.